include/ck/tensor_operation/gpu/device/impl/device_gemm_xdl_cshuffle_v3.hpp

```diff
@@ -21,6 +21,104 @@ namespace ck {
 namespace tensor_operation {
 namespace device {
 
+/// @brief \"Universal\" GEMM operation with SplitK support.
+///
+/// @par Overview
+///         This GEMM operation is carrying out following mathematical equation:
+///         C{M,N} = C_op(A_op(A{M,K}) * B_op(B{K,N}))
+///         Where A, B are input tensors and C is the output tensor. The A/B/C_op are
+///         elementwise operations that could be applied on each tensor respectively.
+///         This operation is called \"universal\" since it provides multiple blockwise GEMM
+///         pipelines optimized for memory-bound, latency and compute-bound cases.
+///
+/// @note   This Kernel implementation supports SplitK algorithm - that is it may be configured
+///         to split the work of K dimension dot product accumulation into multiple work groups.
+///         The partial products of different workgroups are then reduced using AtomicAdd
+///         operation.
+///
+/// @tparam ALayout     A tensor data layout.
+/// @tparam BLayout     B tensor data layout.
+/// @tparam CLayout     C tensor data layout.
+/// @tparam ADataType   A tensor data type.
+/// @tparam BDataType   B tensor data type.
+/// @tparam CDataType   C tensor data type.
+/// @tparam GemmAccDataType The so called "accumulation" data type related to hardware
+///                         matrix-multiplication instrucion.
+/// @tparam CShuffleDataType The data type used to store matrix-multiplication results into
+///                          LDS memory during \"CShuffle\" data layout optimization.
+/// @tparam AElementwiseOperation An elementwise operation which could by applied on A input tensor
```

Comment:
```suggestion
/// @tparam AElementwiseOperation Elementwise operation applied to the A input tensor
```
Unless it's only applied in some cases. In which case, it should be specified.